notebooks/02-ingestion/12-spatial-dimensions.ipynb

```diff
@@ -0,0 +1,709 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "0cb3b009-ebde-4d56-9d59-a028d66d8309",
+   "metadata": {},
+   "source": [
+    "# Ingestion and query of spatial dimensions\n",
+    "<!--\n",
+    "  ~ Licensed to the Apache Software Foundation (ASF) under one\n",
+    "  ~ or more contributor license agreements.  See the NOTICE file\n",
+    "  ~ distributed with this work for additional information\n",
+    "  ~ regarding copyright ownership.  The ASF licenses this file\n",
+    "  ~ to you under the Apache License, Version 2.0 (the\n",
+    "  ~ \"License\"); you may not use this file except in compliance\n",
+    "  ~ with the License.  You may obtain a copy of the License at\n",
+    "  ~\n",
+    "  ~   http://www.apache.org/licenses/LICENSE-2.0\n",
+    "  ~\n",
+    "  ~ Unless required by applicable law or agreed to in writing,\n",
+    "  ~ software distributed under the License is distributed on an\n",
+    "  ~ \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY\n",
+    "  ~ KIND, either express or implied.  See the License for the\n",
+    "  ~ specific language governing permissions and limitations\n",
+    "  ~ under the License.\n",
+    "  -->\n",
+    "\n",
+    "This notebook demonstrates the spatial dimension capabilities in Druid.\n",
+    "You will find examples that: \n",
+    "- Ingest spatial data\n",
+    "- Query spatial dimensions\n",
+    "- Use spatial filters to efficiently find events within a rectangle, a radius or a polygon shape  \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bbdbf6ad-ca7b-40f5-8ca3-1070f4a3ee42",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "This tutorial works with Druid 29.0.0 or later.\n",
+    "\n",
+    "#### Run with Docker\n",
+    "\n",
+    "<!-- Profiles are:\n",
+    "`druid-jupyter` - just Jupyter and Druid\n",
+    "`all-services` - includes Jupyter, Druid, and Kafka\n",
+    " -->\n",
+    "\n",
+    "Launch this tutorial and all prerequisites using the `all-services` profile of the Docker Compose file for Jupyter-based Druid tutorials. For more information, see the Learn Druid repository [readme](https://github.com/implydata/learn-druid).\n",
+    "   \n",
+    "   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5007a243-b81a-4601-8f57-5b14940abbff",
+   "metadata": {},
+   "source": [
+    "## Initialization\n",
+    "\n",
+    "The following cells set up the notebook and learning environment ready for use.\n",
+    "\n",
+    "### Set up and connect to the learning environment\n",
+    "\n",
+    "Run the next cell to set up the Druid Python client's connection to Apache Druid.\n",
+    "\n",
+    "If successful, the Druid version number will be shown in the output."
+   ]
+  },
+  {
```

Comment:
If Kafka is not required, can we use this code instead?

> import druidapi
> import os
> 
> if 'DRUID_HOST' not in os.environ.keys():
>     druid_host=f"http://localhost:8888"
> else:
>     druid_host=f"http://{os.environ['DRUID_HOST']}:8888"
>     
> print(f"Opening a connection to {druid_host}.")
> druid = druidapi.jupyter_client(druid_host)
> 
> display = druid.display
> sql_client = druid.sql
> status_client = druid.status
> 
> status_client.version

Reply:
done. I left the rest_client which is needed and I'm using display_client instead of display because it overrides the "display" object in Jupyter.